lib/grape/dsl/request_response.rb

```diff
@@ -104,7 +104,7 @@ def rescue_from(*args, &block)
             handler = block
           end
 
-          options = args.extract_options!
+          options = args.extract_options!.dup
```

Comment:
So what's wrong with just checking `options[:with]` below? Why does it need to `delete` that value?